### DISCUSSION TITLE
Install Source showing incorrectly for apps installed in multiple profiles

### DISCUSSION THREAD

**Discussion by TheRealMystic** [d4rken-org/permission-pilot#257] [Q&A]
In the attached screenshot, you'll see that the File Manager+ app is showing up along with details of its Twin underneath. The problem here is the install source shown for the app installed in the Main Profile. In this case, it should be SAI, and not Shelter, because Shelter is the install source for the app installed in the Work Profile.

![Screenshot_2022-07-19-18-52-26-64_ce34de6323cfaeb7a2880d42987990b1](https://user-images.githubusercontent.com/54448474/179761776-3160e598-8612-43cf-88fc-4ee27ef4635e.jpg)

**Comment by d4rken** [comment]
This feels like a ROM bug and working as intended from the app side, we ask the system and this is what we are told.
We can further look into this after #40 by checking a debug log from your device.

I've noticed that we don't have access to all information of the new API, as it requires the app to hold additional permission:
https://developer.android.com/reference/android/content/pm/InstallSourceInfo

> Note that the InstallSourceInfo returned by [PackageManager#getInstallSourceInfo(String)](https://developer.android.com/reference/android/content/pm/PackageManager#getInstallSourceInfo(java.lang.String)) will not have this information available unless the calling application holds the INSTALL_PACKAGES permission.

Would it be worth it asking for this permission to show this? :thinking:

**Comment by TheRealMystic** [comment]
> This feels like a ROM bug and working as intended from the app side, we ask the system and this is what we are told. We can further look into this after #40 by checking a debug log from your device.
> 
Okay. Sure.


> I've noticed that we don't have access to all information of the new API, as it requires the app to hold additional permission: https://developer.android.com/reference/android/content/pm/InstallSourceInfo
> 
> > Note that the InstallSourceInfo returned by [PackageManager#getInstallSourceInfo(String)](https://developer.android.com/reference/android/content/pm/PackageManager#getInstallSourceInfo(java.lang.String)) will not have this information available unless the calling application holds the INSTALL_PACKAGES permission.
> 
> Would it be worth it asking for this permission to show this? 🤔
>
I'm not really in a position to answer this. You are better placed to decide on this.
